sdk/demos/intel/js/renderer.js

```diff
@@ -0,0 +1,236 @@
+          gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_WRAP_S, gl.CLAMP_TO_EDGE);
+          gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_WRAP_T, gl.CLAMP_TO_EDGE);
+        } else {
+          if (this.ext !== null) {
+            gl.bindTexture(ext.TEXTURE_VIDEO_IMAGE, self.texture);
+            ext.VideoElementTargetVideoTexture(ext.TEXTURE_VIDEO_IMAGE, video);
+            gl.bindTexture(ext.TEXTURE_VIDEO_IMAGE, null);
+          }
+        }
+
+        resolve(self.videoElement);
+      });
+
+      video.addEventListener('error', function(ev) {
+        console.log(video.error);
+        reject(video.error);
+      }, false);
+
+      video.loop = true;
+      video.autoplay = true;
+      video.muted = true;
+      video.crossOrigin = 'anonymous';
+      video.setAttribute('webkit-playsinline', '');
+      video.src = url;
+      var btn = document.createElement("BUTTON");
+      var t = document.createTextNode("CLICK ME");
+      btn.style.position = "absolute";
+      btn.style.zIndex = "999";
+      btn.style.left = "80";
+      btn.onclick = function() {
+        video.play();
+      }
+      btn.appendChild(t);
+      document.body.appendChild(btn);
+    });
+  };
+
+  Renderer.prototype.play = function() {
+    if (this.videoElement)
+      this.videoElement.play();
+  };
+
+  Renderer.prototype.pause = function() {
+    if (this.videoElement)
+      this.videoElement.pause();
+  };
+
+  Renderer.prototype.isPaused = function() {
+    if (this.videoElement)
+      return this.videoElement.paused;
+    return false;
+  };
+
+  Renderer.prototype.render = function (width, height) {
+    var gl = this.gl;
+    var program = this.program;
+    var using_WEBGL_video_texture = this.enable_WEBGL_video_texture;
+    var ext = this.ext;
+
+    if (!this.videoElement)
+      return;
+
+    gl.useProgram(this.program);
+
+    gl.bindBuffer(gl.ARRAY_BUFFER, this.vertBuffer);
+
+    var position = gl.getAttribLocation(this.program, "position");
+    var texCoord = gl.getAttribLocation(this.program, "texCoord");
+
+    gl.enableVertexAttribArray(position);
+    gl.enableVertexAttribArray(texCoord);
+
+    gl.vertexAttribPointer(position, 3, gl.FLOAT, false, 20, 0);
+    gl.vertexAttribPointer(texCoord, 2, gl.FLOAT, false, 20, 12);
+
+    gl.bindBuffer(gl.ELEMENT_ARRAY_BUFFER, this.indexBuffer);
+
+    gl.activeTexture(gl.TEXTURE0);
+    var diffuse = gl.getUniformLocation(this.program, "diffuse");
+    gl.uniform1i(diffuse, 0);
+
+    if (!using_WEBGL_video_texture) {
+      gl.bindTexture(gl.TEXTURE_2D, this.texture);
+
+      if (this.videoElement && !this.videoElement.paused) {
+        gl.texImage2D(gl.TEXTURE_2D, 0, gl.RGBA, gl.RGBA, gl.UNSIGNED_BYTE, this.videoElement);
+      }
+    } else {
+      gl.bindTexture(ext.TEXTURE_VIDEO_IMAGE, this.texture);
+
+      if (this.videoElement && !this.videoElement.paused && this.ext !== null) {
+        ext.VideoElementTargetVideoTexture(ext.TEXTURE_VIDEO_IMAGE, this.videoElement);
```

Comment:
Similarly, should this be using shareVideoImageWEBGL, with releaseVideoImageWEBGL after the call to drawElements?

Reply:
The comment on releaseVideoImageWEBGL:
Quick try and find that I missed implementing this API in blink now.... Will do a quick update.